File: trunk/devops/assist/mysql-source-filter/src/main/java/com/netx/data/filter/message/SourceChangeMessage.java
package com.netx.data.filter.message;

import com.alibaba.otter.canal.protocol.CanalEntry;

import java.util.Date;

/**
 * Created by czou on 5/7/2018.
 */
public class SourceChangeMessage {
    private String dbName;
    private String tableName;
    private String pkId;
    private CanalEntry.EventType eventType;
    private Date createTime;
    private Date updateTime;
    private Boolean deleted;

    public String getDbName() {
        return dbName;
    }

    public void setDbName(String dbName) {
        this.dbName = dbName;
    }

    public String getTableName() {
        return tableName;
    }

    public void setTableName(String tableName) {
        this.tableName = tableName;
    }

    public String getPkId() {
        return pkId;
    }

    public void setPkId(String pkId) {
        this.pkId = pkId;
    }

    public CanalEntry.EventType getEventType() {
        return eventType;
    }

    public void setEventType(CanalEntry.EventType eventType) {
        this.eventType = eventType;
    }

    public Date getCreateTime() {
        return createTime;
    }

    public void setCreateTime(Date createTime) {
        this.createTime = createTime;
    }

    public Date getUpdateTime() {
        return updateTime;
    }

    public void setUpdateTime(Date updateTime) {
        this.updateTime = updateTime;
    }


    public Boolean getDeleted() {
        return deleted;
    }

    public void setDeleted(Boolean deleted) {
        this.deleted = deleted;
    }
}

File: trunk/devops/nx_data/nx_source_sync/src/main/scala/com/netx/message/SourceChangeMessage.scala
package com.netx.message

import com.alibaba.otter.canal.protocol.CanalEntry.EventType
import io.circe.{ Decoder, Encoder }
import io.circe.generic.semiauto.{ deriveDecoder, deriveEncoder }

object SourceChangeMessage {
  implicit val decodeSCMessage: Decoder[SourceChangeMessage] = deriveDecoder[SourceChangeMessage]
  implicit val encodeSCMessage: Encoder[SourceChangeMessage] = deriveEncoder[SourceChangeMessage]

  implicit val decodeEventType: Decoder[EventType] = Decoder[String].emap {
    case "UPDATE" => Right(EventType.UPDATE)
    case "INSERT" => Right(EventType.INSERT)
    case "DELETE" => Right(EventType.DELETE)
    case _ => Right(EventType.QUERY)
  }
  implicit val encodeEventType: Encoder[EventType] = Encoder[String].contramap(_.toString)
}

case class SourceChangeMessage(dbName: String, tableName: String, eventType: EventType, pkId: Option[String], relationId: Option[String],
  createTime: Option[String], updateTime: Option[String], deleted: Option[Boolean])
